frontend/src/components/readsets/ReadsetsListContent.tsx

```diff
@@ -1,105 +1,279 @@
+            </Button>
+            <Button
+                style={{ margin: 6 }}
+                onClick={async () => {
+                    await releaseStatusManager.updateReleaseStatus(filters)
+                    await refreshDataset()
+                }}
+                type={"primary"}
+                disabled={!saveChangesEnabled}>
+                Save Changes
+            </Button>
+        </div>
+    },
+    [canReleaseOrBlockReadsets, filters, refreshDataset, releaseStatusManager])
+
+    const mapContainerIDs = useItemsByIDToDataObjects(selectReadsetsByID, wrapReadset)
+
+    return <div>
+        <AppPageHeader title="Readsets" />
+        <PageContent>
+            {Object.keys(filters).length > 0 &&
+                <FilterPanel
+                    descriptions={[]}
+                    filters={filters}
+                    setFilter={readsetTableCallbacks.setFilterCallback}
+                    setFilterOption={readsetTableCallbacks.setFilterOptionsCallback}
+                />
             }
-        }
-    }
+            <div style={{
+                display: 'flex',
+                justifyContent: 'space-between'
+            }}>
+                {extraButtons}
+                <FiltersBar filters={filters} clearFilters={readsetTableCallbacks.clearFiltersCallback} />
+            </div>
+            <PagedItemsTable<ObjectWithReadset>
+                columns={columns}
+                expandable={expandableMetricConfig}
+                getDataObjectsByID={mapContainerIDs}
+                pagedItems={readsetTableState}
+                usingFilters={false}
+                {...readsetTableCallbacks}
+                initialLoad={false}
+            />
+        </PageContent>
+    </div>
+}
+
+export default ReadsetsListContent
 
-    const [releaseStatusOption, dispatchReleaseStatusOption] = useReducer(
-        releaseStatusOptionReducer,
+
+
+function useReleaseStatusManager(datasetID: Dataset["id"]) {
+    const [readsetReleaseStates, setReadsetReleaseStates] = useState<
+        Record<Readset["id"],
         {
-            all: undefined,
-            specific: {},
+            old: ReleaseStatus,
+            new: ReleaseStatus.RELEASED | ReleaseStatus.BLOCKED | undefined
+        } | undefined
+    >>({})
+    const dispatch = useAppDispatch()
+    useEffect(() => {
+        dispatch(api.readsets.list({ dataset__id__in: datasetID, limit: 10000 })).then((readsets) => {
+            const newReadsetReleaseStates = readsets.data.results.reduce<typeof readsetReleaseStates>((acc, readset) => {
+                acc[readset.id] = { old: readset.release_status, new: undefined }
+                return acc
+            }, {})
+            setReadsetReleaseStates(newReadsetReleaseStates)
+        })
+    }, [datasetID, dispatch])
+
+    const setAllReleaseStatus = useCallback((newReleaseStatus: ReleaseStatus.RELEASED | ReleaseStatus.BLOCKED) => {
+        setReadsetReleaseStates(produce((prev) => {
```

Comment:
Don't you need readsetReleaseStates as first param for produce ?

Reply:
That's just a variable name. It can be any name. But yeah I agree that `readsetReleaseStates` would be a better name.

Reply:
The question is not about the name it is just that i was under the impression the function produce took 2 params the first being the immutable state.